typing/typecore.ml

```diff
@@ -5532,6 +5541,36 @@ let report_unification_error ~loc ?sub env err
       ?type_expected_explanation txt1 txt2
   ) ()
 
+let report_this_function ppf funct =
+  if Typedtree.exp_is_short funct then
+    let pexp = Untypeast.untype_expression funct in
+    Format.fprintf ppf "The function '%a'" Pprintast.expression pexp
+  else Format.fprintf ppf "This function"
+
+let report_too_many_arg_error ~funct ~func_ty ~previous_arg_loc
+    ~extra_arg_loc loc =
+  let open Location in
+  let app_loc =
+    (* Span the application, including the extra argument. *)
+    { loc_start = loc.loc_start;
+      loc_end = extra_arg_loc.loc_end;
+      loc_ghost = false }
+  and tail_loc =
+    (* Possible location for a ';'. *)
+    let loc_start = previous_arg_loc.loc_end in
+    { loc_start;
+      loc_end = { loc_start with pos_cnum = loc_start.pos_cnum + 1 };
+      loc_ghost = false }
+  in
+  let sub = [
+    msg ~loc:tail_loc "Hint: Did you forget a ';'?";
+    msg ~loc:extra_arg_loc "This argument is not expected.";
```

Comment:
Note that the functor error message uses `This extra argument` to emphasize that the argument doesn't fit the function type. I think could reuse the term here.

Reply:
Done.